src/Game/Systems/PlayerSpawnSystem.cpp

```diff
@@ -136,27 +134,38 @@ bool PlayerSpawnSystem::OnInputCommand(Events::InputCommand& e)
     auto iter = m_SpawnRequests.begin();
     for (; iter != m_SpawnRequests.end(); ++iter) {
         if (iter->PlayerID == e.PlayerID) {
-            // If player wants to switch class, remove their spawn request.
-            if (e.Command == "SwapToClassPick") {
-                m_SpawnRequests.erase(iter);
+            // If player wants to switch team or class , remove their selected class so they don't spawn.
+            if (e.Command == "SwapToTeamPick" || e.Command == "SwapToClassPick") {
+                iter->Class = -1;
+                return true;
             }
             break;
         }
     }
 
-    if (e.Command == "SwapToClassPick") {
-        return true;
-    }
+    //If we get here we got a PickTeam or PickClass, so add or alter a spawn request.
 
     if (iter != m_SpawnRequests.end()) {
-        // If player is in queue to spawn, then change their team affiliation in the request.
-        iter->Team = (ComponentInfo::EnumType)e.Value;
+        // If player is in queue to spawn, then change their team affiliation or class in the request.
+        if (e.Command == "PickTeam") {
+            iter->Team = (ComponentInfo::EnumType)e.Value;
+        } else {
+            iter->Class = (ComponentInfo::EnumType)e.Value;
+        }
     } else if (m_PlayerEntities.count(e.PlayerID) == 0 || !m_PlayerEntities[e.PlayerID].Valid()) {
         // If player is not in queue to spawn, then create a spawn request, 
         // but only if they are spectating and/or just connected.
         SpawnRequest req;
         req.PlayerID = e.PlayerID;
-        req.Team = (ComponentInfo::EnumType)e.Value;
+        if (e.Command == "PickTeam") {
+            req.Team = (ComponentInfo::EnumType)e.Value;
+            req.Class = -1;         // TODO: -1 Signifies no class picked, enum here later?
+        } else {
+            // Should never get here, since you should have picked a team before you ever get a chance to pick class.
+            LOG_WARNING("Sequence error: Should not be able to pick class before team");
+            req.Team = 1;           // TODO: 1 Signifies spectator, should probably have real enum here later.
+            req.Class = (ComponentInfo::EnumType)e.Value;
```

Comment:
If you somehow end up here, will you spawn as a [Class] spectator? If so this
> // A spectator can't die anyway
>       if ((ComponentInfo::EnumType)cTeam["Team"] == cTeam["Team"].Enum("Spectator")) {
>           return false;
>       }
might be a bad check?